src/analysis/retail/paladin/holy/modules/dragonflight/tier/T30TierSet.tsx

```diff
@@ -0,0 +1,136 @@
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
+import { TALENTS_PALADIN } from 'common/TALENTS';
+import { formatNumber } from 'common/format';
+import HIT_TYPES from 'game/HIT_TYPES';
+import { getLightsHammerHeals } from '../../../normalizers/CastLinkNormalizer';
+import { SpellLink } from 'interface';
+
+const CRIT_HEAL_AMOUNT = 1.65;
+const TWO_PIECE_HEAL_INC = CRIT_HEAL_AMOUNT / 1.5; // 210% heal instead of 150% on crit
+const HOLY_PRISM_INC = 0.8;
+
+class T30HpalTierSet extends Analyzer {
+  has4Piece: boolean = false;
+  hsHealing: number = 0;
+  hsOverhealing: number = 0;
+  hpGained: number = 0;
+  hpWasted: number = 0;
+  lhTiktok: boolean = false; // if true, count set of lh heals
+  fourPcHealing: number = 0;
+  fourPcOverhealing: number = 0;
+  countedHeals: Set<HealEvent> = new Set();
+  extraLhHits: number = 0;
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.has2PieceByTier(TIERS.T30);
+    if (!this.active) {
+      return;
+    }
+    this.has4Piece = this.selectedCombatant.has4PieceByTier(TIERS.T30);
+    this.addEventListener(
+      Events.heal.by(SELECTED_PLAYER).spell(SPELLS.HOLY_SHOCK_HEAL),
+      this.onHolyShockHeal,
+    );
+    this.addEventListener(
+      Events.heal
+        .by(SELECTED_PLAYER)
+        .spell([SPELLS.HOLY_PRISM_HEAL, SPELLS.HOLY_PRISM_HEAL_DIRECT]),
+      this.onHolyPrismHeal,
+    );
+    this.addEventListener(
+      Events.heal.by(SELECTED_PLAYER).spell(SPELLS.LIGHTS_HAMMER_HEAL),
+      this.onLHHeal,
+    );
+    this.addEventListener(Events.resourcechange.by(SELECTED_PLAYER), this.onHpGain);
+  }
+
+  onHolyShockHeal(event: HealEvent) {
+    if (event.hitType !== HIT_TYPES.CRIT) {
+      return;
+    }
+    this.hsHealing += calculateEffectiveHealing(event, TWO_PIECE_HEAL_INC);
+    this.hsOverhealing += calculateOverhealing(event, TWO_PIECE_HEAL_INC);
+  }
+
+  onHpGain(event: ResourceChangeEvent) {
+    if (
+      event.ability.guid === TALENTS_PALADIN.LIGHTS_HAMMER_TALENT.id ||
+      event.ability.guid === TALENTS_PALADIN.HOLY_PRISM_TALENT.id
+    ) {
+      this.hpGained += event.resourceChange;
+      this.hpWasted += event.waste;
+    }
+  }
+
+  onHolyPrismHeal(event: HealEvent) {
+    this.fourPcHealing + calculateEffectiveHealing(event, HOLY_PRISM_INC);
+    this.fourPcOverhealing + calculateOverhealing(event, HOLY_PRISM_INC);
+  }
+
+  onLHHeal(event: HealEvent) {
+    if (this.countedHeals.has(event)) {
+      return;
+    }
+    const healEvents = getLightsHammerHeals(event);
+    if (this.lhTiktok) {
+      healEvents.forEach((ev) => {
+        this.fourPcHealing += ev.amount;
+        this.fourPcOverhealing += ev.overheal || 0;
+        this.extraLhHits += 1;
+      });
+    }
+    // mark events as counted
+    healEvents.forEach((ev) => {
+      this.countedHeals.add(ev);
+    });
+    this.lhTiktok = !this.lhTiktok;
```

Comment:
i do like how simple this approach is. but i am a bit worried how it behaves if the raid has to move in and out of the area and misses some ticks. i think we should reset `this.lhTiktok = false` when we see a new cast event for light's hammer.

an alternative approach could be based on the timing of the ticks. since the tick rate doesn't look hasted maybe this isn't too complicated?

Reply:
Correct, LH ticks are not hasted so you could just add a timestamp check with some ms threshold to protect the integrity of your clock